filesystems/blob/src/main/scala/cromwell/filesystems/blob/WorkspaceManagerApiClientProvider.scala

```diff
@@ -22,9 +31,80 @@ class HttpWorkspaceManagerClientProvider(baseWorkspaceManagerUrl: WorkspaceManag
     client
   }
 
-  def getControlledAzureResourceApi(token: String): ControlledAzureResourceApi = {
+  def getWorkspaceApi(token: String): WsmWorkspaceApi = {
+    val apiClient = getApiClient
+    apiClient.setAccessToken(token)
+    WsmWorkspaceApi(new WorkspaceApi(apiClient))
+  }
+
+  def getResourceApi(token: String): WsmResourceApi = {
+    val apiClient = getApiClient
+    apiClient.setAccessToken(token)
+    WsmResourceApi(new ResourceApi(apiClient))
+  }
+
+  def getLandingZonesApi(token: String): WsmLandingZonesApi = {
+    val apiClient = getApiClient
+    apiClient.setAccessToken(token)
+    WsmLandingZonesApi(new LandingZonesApi(apiClient))
+  }
+
+  def getControlledAzureResourceApi(token: String): WsmControlledAzureResourceApi = {
     val apiClient = getApiClient
     apiClient.setAccessToken(token)
-    new ControlledAzureResourceApi(apiClient)
+    WsmControlledAzureResourceApi(new ControlledAzureResourceApi(apiClient))
+  }
+}
+
+case class WsmLandingZonesApi(landingZonesApi : LandingZonesApi) {
+  def findLandingZoneForBillingProfile(billingProfileId: UUID): Try[UUID] = {
+    for {
+      landingZones <- Try(landingZonesApi.listAzureLandingZones(billingProfileId).getLandingzones())
+      landingZoneOption = landingZones.asScala.find(_ => true) // Find the first expecting that there should be only one
+      landingZone <- landingZoneOption.map(_.getLandingZoneId()).toRight(new Exception("Billing profile not found")).toTry
+    } yield landingZone
+  }
+
+  def findStorageAccountForLandingZone(landingZoneId: UUID): Try[StorageAccountName] = {
+    for {
+      resources <- Try(landingZonesApi.listAzureLandingZoneResources(landingZoneId).getResources())
+      sharedResourcesOption = resources.asScala.find(pg => pg.getPurpose() == "SHARED_RESOURCE")
+      sharedResources <- sharedResourcesOption.toRight(new Exception("No shared resources found")).toTry
```

Comment:
I think I'd like to update these error messages for a bit more detail if we're going to keep this.